docs/reference.md

```diff
@@ -131,3 +131,57 @@ dedicated test-suite, consisting of several tests exercising various runtime
 behaviors, for each StableHLO Op. The tests can be found
 [here](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/) (e.g.
 interpret\_\*.mlir).
+
+### Testing guidelines
+
+**(G1) Do we need to test for all the supported types for every op ?**
+
+We can use the following thumb rule to decide it:
+
+While implementing an op, if we need to introduce code in the corresponding
+`eval` function to handle types differently, then it is imperative to have tests
+to cover for all those types. But in cases where a set of types are handled
+uniformly, like for `add` op, all the supported integer types are handled alike
+using `llvm::APInt` APIs, then we can skip adding tests for all those types and,
+instead, add a single test with a type, with maximum bit-width (an arbitrary
+choice), from the set.
```

Comment:
I'm not sure I exactly follow how many tests we need for `add`? Just one? One per type category (i.e. one for ints, one for floats, etc)? By the way, I really like like the idea of not writing tests for llvm::APInt and llvm::APFloat - but let's document this in a bit more detail.